scompose/project/project.py

```diff
@@ -411,50 +434,38 @@ def _create(
             # Generate shared hosts file
             hosts_file = self.create_hosts(lookup)
 
-        # First create those with no dependencies
-        while names:
-
-            for instance in self.iter_instances(names):
-
-                # Flag to indicated create
-                do_create = True
-
-                # Ensure created, skip over if not
-                depends_on = instance.params.get("depends_on", [])
-                for depends_on in depends_on:
-                    if depends_on not in created:
-                        count += 1
-                        do_create = False
-
-                if do_create:
-
-                    # Generate a resolv.conf to bind to the container
-                    if self.sudo and not no_resolv:
-                        resolv = self.generate_resolv_conf()
-                        instance.volumes.append("%s:/etc/resolv.conf" % resolv)
-
-                        # Create a hosts file for the instance based, add as volume
-                        instance.volumes.append("%s:/etc/hosts" % hosts_file)
+        for instance in self.iter_instances(names):
+            depends_on = instance.params.get("depends_on", [])
+            for dep in depends_on:
+                if dep not in created:
+                    circular_dep = True
+
+            # Generate a resolv.conf to bind to the container
+            if self.sudo and not no_resolv:
+                resolv = self.generate_resolv_conf()
+                instance.volumes.append("%s:/etc/resolv.conf" % resolv)
+
+                # Create a hosts file for the instance based, add as volume
+                instance.volumes.append("%s:/etc/hosts" % hosts_file)
+
+            # If we get here, execute command and add to list
+            create_func = getattr(instance, command)
```

Comment:
Could you walk me through the logic of this code block?

Reply:
It was already like that, I just reformatted the spacing once deleted the outer loop.

I looked into it, the passed `command` is the name of the method to execute on the `instance`: can be either `create` (in case of build) or simply `up` (in case of run)

Reply:
Ah okay, what I was curious about is how the logic works with the deleted outer loop. For example, the do_create variable is gone, If there is a way to test this and make sure that the functionality has not changed, then that should be enough validation for approval.